server/commands/deployments/DeployService.js

```diff
@@ -36,6 +42,39 @@ module.exports = function DeployServiceCommandHandler(command) {
     let deployment = yield validateCommandAndCreateDeployment(command);
     let destination = yield packagePathProvider.getS3Path(deployment);
```

Comment:
The `DeployService` command was already pretty comically large - could we create a separate module for checking deployment schedule state and move some of the new logic there?